src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToNewComponentCodeActionProviderTest.cs

```diff
@@ -67,6 +76,128 @@ public async Task Handle_InvalidFileKind()
         Assert.Empty(commandOrCodeActionContainer);
     }
 
+    [Fact]
+    public async Task Handle_InProperMarkup_ReturnsNull()
+    {
+        // Arrange
+        var documentPath = "c:/Test.razor";
+        var contents = """
+            page "/"
+            
+            <PageTitle>Home</PageTitle>
+            
+            <div id="parent">
+                <div>
+                    <h1>Div a title</h1>
+                    <p>Div $$a par</p>
+                </div>
+                <div>
+                    <h1>Div b title</h1>
+                    <p>Div b par</p>
+                </div>
+            </div
+            
+            <h1>Hello, world!</h1>
+            
+            Welcome to your new app.
+            """;
+        TestFileMarkupParser.GetPosition(contents, out contents, out var cursorPosition);
+
+        var request = new VSCodeActionParams()
+        {
+            TextDocument = new VSTextDocumentIdentifier { Uri = new Uri(documentPath) },
+            Range = new Range(),
```

Comment:
Given this is passing in an empty range, does that mean this test isn't actually testing the change in this PR?

Would like to see a bunch more tests for the various scenarios that the logic handles. With a helper method it should be hopefully straight foward. The test markup syntax already supports `[|` and `|]` as selection markers too.

Reply:
+1 to this more tests would be good